syntax = "proto3";

service PredictionService {
  rpc Predict (PredictRequest) returns (PredictResponse);
  rpc Health (Empty) returns (HealthResponse);
}

message PredictRequest {
  float sepal_length = 1;
  float sepal_width = 2;
  float petal_length = 3;
  float petal_width = 4;
}

message PredictResponse {
  string prediction = 1;
  string model_version = 2;
}

message Empty {}

message HealthResponse {
  string status = 1;
}
